pkg/child/child.go

```diff
@@ -90,3 +112,41 @@ func Run(ctx context.Context, cfg *RunConfig) (int, error) {
 
 	return exitCode, nil
 }
+
+// environ compiles the appropriate environment to pass to the child process.
+// The overridden environment is always added, even if not explicitly
+// allowed/denied.
+func environ(osEnv, allowedKeys, deniedKeys, overrideEnv []string) []string {
+	finalEnv := make([]string, 0, len(osEnv)+len(overrideEnv))
+
+	// Select keys that match the allow filter (if given) but not the deny filter.
+	for _, v := range osEnv {
+		k := strings.SplitN(v, "=", 2)[0]
+		if (len(allowedKeys) > 0 && anyGlobMatch(k, allowedKeys)) &&
+			!anyGlobMatch(k, deniedKeys) {
+			finalEnv = append(finalEnv, v)
+		}
+	}
+
+	// Add overrides at the end, after any filtering
+	finalEnv = append(finalEnv, overrideEnv...)
```

Comment:
What happens if overrideEnv keys appear in the user provided env vars?

Reply:
The later keys in the slice take precedence (as expected). That's taken care of by `cmd`. 

That's actually true in unix/linux in general. It's possible to have multiple keys with different values in `env` - it's why env is a slice and not a map.